### ISSUE TITLE
Mouse cursor isn't always displayed with RmlUI

### ISSUE THREAD

**Issue by necessarily-equal** [Unvanquished/Unvanquished#1887]
When clicking to respawn you sometime need to move the cursor around and manage to hover the dialog box before RmlUI starts displaying the cursor. It doesn't happen all the time, and to be fair it's sometimes hard to know if it's just me not finding the cursor.

**Comment by necessarily-equal** [comment]
This is a RmlUI regression

**Comment by slipher** [comment]
How do you produce this? When I click to open the class selection dialog, it always centers the cursor (therefore putting it within the dialog).

**Comment by necessarily-equal** [comment]
This may be SDL_VIDEODRIVER=wayland

**Comment by illwieckz** [comment]
If I'm right some mouse cursor code had to be rewritten when moving to RmlUi, @DolceTriade may know more about that.

I assume this is a blocking issue for release?

**Comment by DolceTriade** [comment]
RmlUI has builtin support for the mouse cursor whereas in librocket it was a decorator.

**Comment by necessarily-equal** [comment]
I don't think this is a release blocker. I've mentioned it in https://github.com/DaemonEngine/Daemon/issues/482 in case it would be wayland-related

**Comment by DolceTriade** [comment]
If this is not a release blocker, I'm moving this to 0.54.0.

**Comment by necessarily-equal** [comment]
@slipher any idea about this one? I had a look but couldn't fix it. I rembember you made a lot of fixes about cursors for the 0.53.

**Comment by slipher** [comment]
According to my previous comment I was unable to reproduce this

**Comment by necessarily-equal** [comment]
I assume you need a wayland compositor for that, and use `SDL_VIDEODRIVER=wayland` to force the wayland backend.

**Comment by necessarily-equal** [comment]
While I'd want to fix this issue, I don't think it being worked on at the moment. If somebody or wants to fix it before 0.54, this could still be possible, but I'm removing it from the todo list for 0.54.

**Comment by DolceTriade** [comment]
@slipher Did you end up rewriting the mouse cursor code? I think this isn't anything I've seen.

**Comment by slipher** [comment]
I don't think were really significant changes since the RmlUi migration. I haven't ever gotten a Wayland environment set up to try it.

**Comment by necessarily-equal** [comment]
@cu-kai and I confirm that this is still a problem.

This happens on Wayland typically at the beginning of a game.

Steps to reproduce (native or 0.55.0 flatpak):

1. `SDL_VIDEODRIVER=wayland unvanquished` or `DISPLAY= flatpak run net.unvanquished.Unvanquished`
2. `/devmap chasm`
3. click to join a team
4. the cursor is gone and reappears when passing over the dialog

Depending where you cursor is from the start it could already be on top of the window already when it opens. In that case you could restart the game to trigger for example. The cursor stays at the same place as on main menu, so you can put it somewhat on the side before starting the game to be sure to trigger the bug.

**Comment by slipher** [comment]
I now have a setup with Wayland and can reproduce this.

With SDL3 (see https://github.com/DaemonEngine/Daemon/pull/1433), it's better although still buggy. When the team selection menu opens, the mouse cursor is displayed in the center of the screen. But apparently that does not reflect its true position. As soon as you start moving the mouse, the cursor teleports to a different position and then it starts to work normally. So it's like centering the mouse doesn't work as expected.

**Comment by illwieckz** [comment]
Here is test unizip with Wayland-enabled SDL3:

- https://dl.illwieckz.net/j/unvanquished_0-20250910-025135.zip

**Comment by illwieckz** [comment]
Quote from yesterday's chat:

```
<cu-kai> illwieckz: i played quite a bit yesterday afternoon with a native SDL3 engine built from master,
all seemed ok to me
```

@necessarily-equal can you test for me the above build on Wayland?

**Comment by necessarily-equal** [comment]
I'll be unfortunately quite busy this weekend and next week, I'll see if I can test later

**Comment by slipher** [comment]
> With SDL3 (see [DaemonEngine/Daemon#1433](https://github.com/DaemonEngine/Daemon/pull/1433)), it's better although still buggy. When the team selection menu opens, the mouse cursor is displayed in the center of the screen. But apparently that does not reflect its true position. As soon as you start moving the mouse, the cursor teleports to a different position and then it starts to work normally. So it's like centering the mouse doesn't work as expected.

I retested and no longer reproduce this. The mouse warping works fine. Maybe that only happened in a buggy early draft of the PR.

**Comment by illwieckz** [comment]
I haven't reproduced it on Wayland either. Let's consider it fixed.
